work_queue/src/work_queue_worker.c

```diff
@@ -521,6 +530,54 @@ int link_recursive( const char *source, const char *target )
 	}
 }
 
+/*
+Untar and unpack the conda environment. If this setup fails, then the task is
+inserted to the table of process running, but immediately marked with an error
+Returns 0 on success, of a pid that can be used to account for the task.
+*/
+static int setup_conda_environment(struct work_queue_process *p) {
+	if(!p->task->conda_environment) {
+		return 0;
+	}
+
+	struct environment *e = hash_table_lookup(environments, p->task->conda_environment);
+	if(!e->error && !e->expansion) {
+		e->expansion = string_format("cache/%s-expanded", p->task->conda_environment);
+
+		char *cmd = string_format(
+				"set -e\n"
+				"mkdir -p %s\n"
+				"/bin/tar -C %s -xf cache/%s\n"
+				"%s/bin/conda-unpack", e->expansion, e->expansion, p->task->conda_environment, e->expansion);
+		int status = system(cmd);
+		free(cmd);
+
+		if(!WIFEXITED(status) || WEXITSTATUS(status)) {
+			debug(D_WQ, "failed to expand environment %s", p->task->conda_environment);
+			e->error = 1;
+		}
+	}
+
+	if(e->error) {
+		/* get a fake pid to insert to table */
```

Comment:
Funny, but effective!
Maybe expand this comment to explain to the casual reader
that yes, this really is necessary, in order to avoid namespace conflicts.

Reply:
What's another process among friends? :)